python/ray/data/tests/test_dataset.py

```diff
@@ -3319,6 +3320,30 @@ def flaky_mapper(x):
         ).take()
 
 
+def test_custom_arrow_array_serializer(ray_start_regular):
+    arr = pa.array(list(range(100)))
+    # Create a zero-copy slice view of arr.
+    view = arr.slice(10, 10)
+    s_arr = pickle.dumps(arr)
+    s_view = pickle.dumps(view)
+    # Check that the slice view is at least twice as small as the full array.
```

Comment:
Can this be a tighter bound? Ideally it seems to be 10%?

Reply:
There appears to be some baseline overhead here, unfortunately. For this particular test case, you get the following serialized sizes:
- `arr` (100 items): 929 bytes
- `view` (10 items): 202 bytes
- `pa.array(list(range(10, 20)))` (10 items): 209 bytes

As you can see here, we're already serializing the 10 item view to a slightly smaller size than the baseline serialization of a fresh 10 item array.